pkg/graveler/staging/manager.go

```diff
@@ -100,3 +113,38 @@ func (m *Manager) DropByPrefix(ctx context.Context, st graveler.StagingToken, pr
 
 	return nil
 }
+
+func (m *Manager) cleanupLoop(ctx context.Context) {
```

Comment:
Add a sleep in the for loop?
Also, If we want an event loop for managers, I think we should do something more generic that can be reused

Reply:
1. We can generalize, if we need to, from the second manager cleanup loop. Trying to accommodate for future use cases is hard.
2. Why adding a sleep?

Reply:
Creating an event loop with a switch case shouldn't be too difficult. You want to add a sleep to allow context switching - if I got what you want to do right, then you don't really need to check wakeup continuously (i.e. its okay to check the value every couple of seconds)
BTW, consider using a buffer of 'events' and switch casing on event type instead of a specific boolean parameter for a specific use case. You can then rename the function to eventListener and reuse it for any other event based actions

Reply:
Done (fixed)